sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/CommentedOutCodeCheck.java

```diff
@@ -62,17 +62,27 @@ protected void scanFile(XmlFile file) {
         continue;
       }
       List<Node> siblingComments = getNextCommentSiblings(comment);
-      if (isParseableXml(siblingComments, charset)) {
-        reportedNodes.addAll(siblingComments);
-        XmlTextRange start = XmlFile.nodeLocation(siblingComments.get(0));
-        XmlTextRange end = XmlFile.nodeLocation(siblingComments.get(siblingComments.size() - 1));
-        reportIssue(new XmlTextRange(start, end), "Remove this commented out code.", Collections.emptyList());
-      }
+      checkCommentBlock(siblingComments, charset);
+      reportedNodes.addAll(siblingComments);
```

Comment:
Is it better to keep `reportedNodes ` to check if a node has already been parsed instead of continuing the loop after the last sibling node?

Reply:
I think I prefer to handle this with the list of `reportedNodes`, rather than playing with index, which can become a mess. From my point of view it makes it a bit easier to read. I would however rename it `visitedNodes`.